### PR TITLE
8.0 - hr_family migration from 7.0

### PR DESCRIPTION
**Pull request by sebalix**

### CODE REVIEW

**Comment by ghost** [comment]
please move each model to a seperate file with the name of the class as filename.

**Comment by max3903** [comment]
@sebalix Are you migrating other modules ? Can you tell me which ones so I can update #86 ?

Thanks!

**Comment by sebalix** [comment]
@max3903 : Not right now, I was just migrating some of them randomly from your list.

@dufresnedavid : okay, I hesitated to do so.

File: hr_family/models/hr_children.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#
+
+from openerp import models, fields
+
+
+class hr_children(models.Model):
+    _name = 'hr.employee.children'
+    _description = 'HR Employee Children'
+
+    name = fields.Char("Name", size=256, required=True)
```
Comment (hbrunn) [inline hr_family/models/hr_children.py]: any reason for size=? You don't need this any more if you don't want to restrict the field's length for whatever reason

File: hr_family/models/hr_employee.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+from openerp import models, fields
+
+
+class hr_employee(models.Model):
+    _name = 'hr.employee'
```
Comment (hbrunn) [inline hr_family/models/hr_employee.py]: this can go away, you're simply inheriting

File: hr_family/models/hr_employee.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    _inherit = 'hr.employee'
+
+    fam_spouse = fields.Char("Name", size=256)
```
Comment (hbrunn) [inline hr_family/models/hr_employee.py]: see comment above

File: hr_family/README.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Employee Family Information
+===========================
+
+This module allows you to enter extra information about employee's family.
+
+Usage
+=====
+
+To use this module, you need to go to on any employee form, and look for the
```
Comment (max3903) [inline hr_family/README.rst]: you need to go to any employee form

File: hr_family/README.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+ * https://www.odoo.com/forum/help-1
+
+Credits
+=======
+
+Contributors
+------------
+
+* Michael Telahun Makonnen <mmakonnen@gmail.com>
```
Comment (max3903) [inline hr_family/README.rst:25]: Please add yourself to the contributors list

File: hr_family/models/hr_children.py (hunk 1 above)
Comment (sebalix) [inline hr_family/models/hr_children.py]: No particular reason, it was declared like that in the branch 7.0 (https://github.com/OCA/hr/blob/7.0/hr_family/hr.py). I will drop all `size=` from these fields.

File: hr_family/models/hr_children.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+from openerp import models, fields
+
+
+class hr_children(models.Model):
+    _name = 'hr.employee.children'
+    _description = 'HR Employee Children'
+
+    name = fields.Char("Name", required=True)
+    dob = fields.Date("Date of Birth")
```
Comment (ghost) [inline hr_family/models/hr_children.py]: Can you change 'dob' to something like date_of_birth ?

File: hr_family/models/hr_employee.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+from openerp import models, fields
+
+
+class hr_employee(models.Model):
```
Comment (ghost) [inline hr_family/models/hr_employee.py]: Please use camel case in your class names

**Comment by ghost** [comment]
@sebalix I put 2 comments, otherwise, LGTM

**Comment by sebalix** [comment]
@dufresnedavid done

**Comment by ghost** [comment]
Thanks! :+1:

**Comment by max3903** [comment]
:+1:

**Pull request merged**